client/state/login/magic-login/reducer.js

```diff
@@ -0,0 +1,156 @@
+	[ SERIALIZE ]: () => false,
+	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
+	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => false,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_ERROR ]: () => false,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_FETCH ]: () => true,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_SUCCESS ]: () => false,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_FETCH ]: () => false,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_RECEIVE ]: () => false,
+} );
+
+export const isFetchingAuth = createReducer( false, {
+	[ DESERIALIZE ]: () => false,
+	[ SERIALIZE ]: () => false,
+	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_FETCH ]: () => true,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_RECEIVE ]: () => false,
+} );
+
+export const emailAddressFormInput = createReducer( '', {
+	[ DESERIALIZE ]: () => '',
+	[ SERIALIZE ]: () => '',
+	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => '',
+	[ MAGIC_LOGIN_SET_INPUT_EMAIL_ADDRESS ]: ( state, { email } ) => email,
+} );
+
+export const emailAddressFormInputIsValid = createReducer( false, {
+	[ DESERIALIZE ]: () => false,
+	[ SERIALIZE ]: () => false,
+	[ MAGIC_LOGIN_SET_INPUT_EMAIL_ADDRESS ]: ( state, { email } ) => {
+		return (
+			typeof email === 'string' &&
+			emailValidator.validate( email )
+		);
+	},
+} );
+
+export const requestEmailError = createReducer( null, {
+	[ DESERIALIZE ]: () => null,
+	[ SERIALIZE ]: () => null,
+	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => null,
+	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => null,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_ERROR ]: ( state, { error } ) => error,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_FETCH ]: () => null,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_SUCCESS ]: () => null,
+	[ MAGIC_LOGIN_SET_INPUT_EMAIL_ADDRESS ]: () => null,
+} );
+
+export const requestedEmailSuccessfully = createReducer( false, {
+	[ DESERIALIZE ]: () => false,
+	[ SERIALIZE ]: () => false,
+	[ MAGIC_LOGIN_HIDE_REQUEST_FORM ]: () => false,
+	[ MAGIC_LOGIN_HIDE_REQUEST_NOTICE ]: () => false,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_ERROR ]: () => false,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_FETCH ]: () => false,
+	[ MAGIC_LOGIN_REQUEST_LOGIN_EMAIL_SUCCESS ]: () => true,
+} );
+
+export const requestAuthError = createReducer( null, {
+	[ DESERIALIZE ]: () => null,
+	[ SERIALIZE ]: () => null,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_FETCH ]: () => null,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_RECEIVE ]: ( state, { error } ) => error,
+} );
+
+export const requestAuthSuccess = createReducer( null, {
+	[ DESERIALIZE ]: () => null,
+	[ SERIALIZE ]: () => null,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_FETCH ]: () => null,
+	[ MAGIC_LOGIN_HANDLE_AUTH_TOKEN_RECEIVE ]: ( state, { status } ) => status
+} );
+
+export default combineReducers( {
+	emailAddressFormInput,
+	emailAddressFormInputIsValid,
+	isFetchingEmail,
+	isShowingCheckYourEmailPage,
+	isShowingExpiredPage,
+	isShowingInterstitialPage,
+	isShowingRequestForm,
+	requestAuthError,
+	requestAuthSuccess,
+	requestEmailError,
+	requestedEmailSuccessfully,
+} );
```

Comment:
Also, if not passed a `schema`, `createReducer` will by default set handlers for `{,DE}SERIALIAZE` to return the reducer's initial state, meaning that for all of the above reducers we should get rid of those lines.